### ISSUE TITLE
Working with other projects?

### ISSUE THREAD

**Issue by TheShadowOfHassen** [olivierkes/manuskript#1223]
A while ago, I was working on a spelling/grammar checker and part of a markdown formatter for GTK3, because all the ones that are on the internet are licensed GNU GPL, and sometimes I don't want to use that license for everything I use. Along the way, I also created my own Language tool wrapped because LanguageToolPython is for GPL. Neither are perfect, but they're at the stage that the only way to make them better is to put them to actual use with a project

So here's what I'm positing, I think both my projects would be useful in Manuskript's GTK port, but I don't want them to be licensed GNU GPL because they can be used for other things as well and some people (including me) don't always want to use GNU GPL (nothing wrong with the GPL license, but I hope people understand that there are reasons).   What I'm wondering is would the manuskript developers be OK with working with a lesser GPL LanguageTool Wrapper and either a MIT or Lesser GPL (I don't know which I'm going to license it with) spelling/grammer/markdown GTK thing for python, and when changes need to be made would the Manuskript people be OK with pushing improvments upstream?

**Comment by TheJackiMonster** [comment]
I mean, Manuskript is licensed under GPL. So if people are contributing to Manuskript, they are most likely agreeing on that license terms. If you want to use specific parts from these contributions to publish them using a different license, you need to ask them for permission individually. Because we don't have any central organization to forward copyright of contributions.

I personally would think that contributions to your wrapper should be done in its own repository anyway. So if people want to improve that, they should go there and contribute regarding its license terms.

**Comment by TheShadowOfHassen** [comment]
Yep, in the own repository. That would work, but for the beginning development, the wrappers might need to be packaged with the gtk port at least until we get everything up and running. People could contribute with the wrappers there and not in the right place, but we can put a note in. It might not be that trickly seeing that we're the ones who are doing the GTK development.

I'll put both projects on Github and then integrate the wrapper with Manuskript gtk.

**Comment by TheShadowOfHassen** [comment]
So here's the LanguageTool Wrapper (that I had to write because if I used LanguageToolPython my programm could only be licensed under the GPL.
 https://github.com/TheShadowOfHassen/Savvy.

I'll touch up the spellchecker wrapper and try to put it up soon. Then I'll Pr it into GTK manuskript.

**Comment by TheShadowOfHassen** [comment]
And the gtk textview extensions: https://github.com/TheShadowOfHassen/gtk-textview-extensions

I haven't gotten any spellchecker other than savvy to work with it, yet. I'll figure that out after I add it to manuskript. (I'll work on the PR tomorrow.)
